src/AccessControl/PasswordHash.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Bolt\AccessControl;
+
+use Bolt\Exception\PasswordHashException;
+use Bolt\Exception\PasswordLegacyHashException;
+
+/**
+ * Password hash management service.
+ *
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+class PasswordHash
```

Comment:
Can we name this something else? It sounds like an entity from first glance, but it's actually a service. What about `PasswordHasher`?